### PR TITLE
Added a auto choose net iface function to the SIL.

### PR DESCRIPTION
**Pull request by SebaLukas**
The first available iface with a link-local IPv6 address is then used.

Signed-off-by: Sebastian Lukas <sebastian.lukas@pionix.de>

### CODE REVIEW

File: modules/JsCarV2G/index.js [hunk 1]
```diff
@@ -66,9 +66,24 @@ function JavaStartedDeferred(mqtt_base_path, module_name, mod) {
...
+        if (element.address.includes('fe80')) {
+          return iface
+        }
+      }
+    }
+  }
+  evlog.warning(`No necessary IPv6 link-local address was found!`)
+  return "eth0"
+}
+
```
Comment (a-w50) [inline modules/JsCarV2G/index.js:84]: Quite some codacy issues, here.  We need to get some setup for the js linting. There is already an .eslint.rc in the root folder of everest core.  Nevertheless, the following code would be more js-like, I guess:
```
function choose_first_ipv6_local() {
  const all_ifs = os.networkInterfaces();

  const found_if = Object.keys(all_ifs).find(
    (iface) => all_ifs[iface].filter((info) => info.address.includes('fe80')).length > 0
  );

  if (found_if) return found_if;

  evlog.warning('No necessary IPv6 link-local address was found!');
  return 'eth0';
}
```

**Review by a-w50 — CHANGES_REQUESTED** [review]

File: modules/JsCarV2G/index.js (hunk 1 above)
Comment (SebaLukas) [reply modules/JsCarV2G/index.js:81]: Tried your suggestion. It is much better.

File: modules/JsRiseV2G/index.js [hunk 2]
```diff
@@ -62,6 +62,19 @@ function JavaStartedDeferred(mqtt_base_path, module_name) {
...
+
+  if (found_if) return found_if;
+
+  evlog.warning('No necessary IPv6 link-local address was found!');
+  return 'eth0';
+}
+
 function check_network_interface(network_iface) {
   let networkfound = false;
   const net_init = os.networkInterfaces();
```
Comment (a-w50) [inline modules/JsRiseV2G/index.js:80]: this you won't need anymore

**Review by a-w50 — APPROVED** [review]

File: modules/JsRiseV2G/index.js (hunk 2 above)
Comment (a-w50) [reply modules/JsRiseV2G/index.js:80]: I mean the net_init var

Comment (a-w50) [reply modules/JsRiseV2G/index.js:80]: if you use single quotes around "auto" codacy should be silent

Comment (SebaLukas) [reply modules/JsRiseV2G/index.js:80]: In the case of a HIL, it should still be checked whether the specified interface is available. Therefore I would leave the check_network_interface function with the net_init var in.

**Pull request merged**